### ISSUE TITLE
Windows 10/ARM support

### ISSUE THREAD

**Issue by dcommander** [libjpeg-turbo/libjpeg-turbo#69]
There is some demand for this, in the context of supporting libjpeg-turbo on Windows phones.  It will require significant effort in order to set up the Windows build system to automatically convert the GNU assembler (GAS) syntax of the current ARM assembly code into ARMASM syntax (fortunately, our old friend gas-preprocessor.pl can do this, although my history with that program suggests that some modifications will likely be required.)  Windows Phone or tablet hardware will be required in order to test and validate the build.

**Comment by CAMOBAP795** [comment]
I had experience with cross-compiling `libjpeg-turbo` for Windows phone 8.1/10. I faces with problems during converting to `ARMASM syntax` with `gas-preprocessor.pl`

Fortunately I was needed only decoding feature, so I solved all problems by some hacks

#### Full list of changes (support decoding only!)

 - Reorder `generate_jsimd_rgb_ycc_convert_neon` macro calls to look like this:

```
     generate_jsimd_rgb_ycc_convert_neon extrgbx, 32, 0, 1, 2
     generate_jsimd_rgb_ycc_convert_neon extbgrx, 32, 2, 1, 0
     generate_jsimd_rgb_ycc_convert_neon extxbgr, 32, 3, 2, 1
     generate_jsimd_rgb_ycc_convert_neon extxrgb, 32, 1, 2, 3
     generate_jsimd_rgb_ycc_convert_neon extrgb,  24, 0, 1, 2
     generate_jsimd_rgb_ycc_convert_neon extbgr,  24, 2, 1, 0
```

 - `GASPP_DEBUG=1 ./gas-preprocessor.pl -force-thumb -as-type armasm -- armasm -c libjpeg-turbo/simd/jsimd_arm_neon.S -o jsimd_arm_neon.o > jsimd_arm_neon.asm`
 - `sed -i.bak s/\.balign\ 16/ALIGN\ 16\;\ replaced\ \.balign\ 16/g jsimd_arm_neon.asm`	- Fix not processed `.balign 16`
 - `sed -i.bak s/\.balign\ 8/ALIGN\ 8\;\ replaced\ \.balign\ 8/g jsimd_arm_neon.asm` - Fix not processed `.balign 8`
 - `sed -i.bak s/INPUT_BUF/r1/g jsimd_arm_neon.asm` - Fix not processed alias in alias `INPUT_BUF` -> r1
 - `sed -i.bak s/OUTPUT_BUF/r2/g jsimd_arm_neon.asm` - Fix not processed alias in alias `OUTPUT_BUF` -> r2  
 - comment all error lines

@dcommander I didn't found any info about how donate for this project, could you please add this info to README or just post it here

**Comment by dcommander** [comment]
Thanks for the info.  I suspected that it would require some effort to convert the ASM code, and given that many Windows developers don't grok Perl, it may be necessary to maintain a statically converted assembly file.  That file could be converted using automated tools such as GASPP, but the conversion would be done by the project maintainer whenever new ARM code is checked in, rather than requiring the end developer to do the conversion.  Since this represents a completely new platform for libjpeg-turbo, I'd need hardware in order to test it, and I'd need funding to pay for the ongoing maintenance and support of the feature.  This would include ensuring that enhancements or fixes to the Linux/Unix ARM code are propagated into the Windows ARM code, maintaining a benchmark spreadsheet for testing performance regressions, etc.  That's why the issue is marked "funding needed."

This enhancement feeds into a proposed enhancement to TurboVNC (https://github.com/TurboVNC/turbovnc/issues/17), one of the projects that pays a lot of my bills.  I posted this issue originally because one of the companies that is building products around TurboVNC expressed interest in a Windows/ARM version of it, which would require implementing a Windows/ARM version of libjpeg-turbo.  However, that was over a year ago, and the landscape for Windows Phone has changed somewhat in the meantime.  Given that fact and the fact that Microsoft's tablets are now using Intel processors, there may simply not be enough demand for a full Windows/ARM version of libjpeg-turbo.

There's a "Donate" button on http://libjpeg-turbo.org.  And thanks in advance for the donation!
